src/detector.c

```diff
@@ -1253,6 +1253,12 @@ float validate_detector_map(char *datacfg, char *cfgfile, char *weightfile, floa
 
                 avg_precision += delta_recall * last_precision;
             }
+            //add remaining area of PR curve when recall isn't 0 at rank-1
+            if(pr[i][0].recall!=0)
+            {
+                //no need of delta_recall=last_recall-0;
+                avg_precision += last_recall * last_precision;
```

Comment:
So actually:
```cpp
            if(pr[i][0].recall!=0)
            {
                //no need of delta_recall=last_recall-0;
                avg_precision += last_recall * last_precision;
             }
```
is equal to 
```cpp
delta_recall = last_recall - 0;
avg_precision += delta_recall * last_precision;
```

Reply:
yes, I added that comment for the same purpose...to keep the meaning intact, but reduce computation.

Reply:
keep that if(pr[i][0].recall!=0) line